Interpreting-The-Spec.md

```diff
@@ -0,0 +1,259 @@
+### Interpreting The OpenAPI Specification Files
+
+#### Spec FIle structure
+
+The root spec file is located in `spec/openapi.yml` and is responsible for tying together the components in each of the other files. It contains a `/paths` section wich declares (and defines through references) all of the API's paths. It also contains some common `/components` which are reused in several of the files. This file also includes a `/tags` section which defines how the spec is organized. Each of the spec files (except the main `openapi.yml`) contains operations which all fall under the same tag and filename. So all endpoints tagged under `secrets` are in the `spec/secrets.yml` file.
+
+#### The $ref keyword
+
+The `$ref` keyword is often used in the Conjur OpenAPI spec in order to reference external YAML objects. This is done in order to reduce duplication and allow the spec to be split across multiple files. The syntax is shown below:
+
+```yaml
+$ref: "<file-path>#<object-path>"
+```
+
+ `<file-path>`: The relative path of a file to reference. Can be omitted entirely to reference objects in the current file.
+
+`<object-path>`: A forward slash separated path to the referenced object where `/` is the root of the document
+
+##### Example: 
```

Comment:
This example uses `#####Example`, then next one uses `e.g.`, and after that examples have no label